src/tabs/tabs.js

```diff
@@ -11,7 +11,8 @@ angular.module('ui.bootstrap.tabs', [])
       var previousSelected = ctrl.tabs[previousIndex];
       if (previousSelected) {
         previousSelected.tab.onDeselect({
-          $event: evt
+          $event: evt,
+          selectedTabIndex: index
```

Comment:
I think using the key `$selectedIndex` would be better, just to emphasize that it is a special value provided by UI Bootstrap.